pkg/kv/bulk/sst_batcher.go

```diff
@@ -285,7 +286,14 @@ func (b *SSTBatcher) flushIfNeeded(ctx context.Context, nextKey roachpb.Key) err
 
 // Flush sends the current batch, if any.
 func (b *SSTBatcher) Flush(ctx context.Context) error {
-	return b.doFlush(ctx, manualFlush, nil)
+	if err := b.doFlush(ctx, manualFlush, nil); err != nil {
+		return err
+	}
+	if !b.maxWriteTS.IsEmpty() {
+		log.VEventf(ctx, 1, "waiting until max write time %s", b.maxWriteTS)
+		return b.db.Clock().SleepUntil(ctx, b.maxWriteTS)
```

Comment:
Any sense in clearing maxWriteTS so that if we call Flush multiple times we don't get the above event?  (I think this code is correct as-is since a second call to SleepUntil on the same maxWriteTS would return immediately).

Reply:
I don't think we typically call `Flush()` multiple times without adding at least one sst between, and if a prior sst set max ts then all of them will I believe, so in practice I doubt clearing just in flush matters.

I did consider comparing it to the clock only event'ing and then sleeping if it was actually in the future, but decided it didn't seem worth it. Perhaps SleepUntil itself should eventf though, since it is probably almost always interesting to see if someone is intentionally sleeping and it already calculates the duration and everything that you might want to see in that event.

Reply:
I think I actually will probably end up coming back here and checking the clock myself before delegating to SleepUntil in a followup, just so we can accumulate the total wait time in our flushStats for reporting, as well as average age of batchTS before we send the SST and % of batchTS that end up pushed (by comparing this ts to the one we sent with). But I'm going to do all that in a follow-up I think, after a followup change to use @erikgrinaker's new SSTTimestamp param that wants to be made on top of this change I think, so merging this one just as-is for now.